src/vr.zig

```diff
@@ -244,6 +245,28 @@ pub const Header = packed struct {
         return null;
     }
 
+    /// Returns whether the immediate sender is a replica or client (if this can be determined).
+    /// Some commands such as .request or .prepare may be forwarded on to other replicas so that
+    /// Header.replica or Header.client only identifies the ultimate origin, not the latest peer.
+    pub fn peer_type(self: *const Header) ?enum { replica, client } {
```

Comment:
Making the return type `enum {replica, client, unknown}` instead would be more explicit and make using this function nicer IMO.